app/post.go

```diff
@@ -1340,6 +1340,64 @@ func (a *App) MaxPostSize() int {
 	return a.Srv().MaxPostSize()
 }
 
+// countThreadMentions returns the number of times the user is mentioned in a specified thread after the timestamp.
+func (a *App) countThreadMentions(user *model.User, post *model.Post, teamId string, timestamp int64) (int64, *model.AppError) {
+	team, err := a.GetTeam(teamId)
+	if err != nil {
+		return 0, err
+	}
+	channel, err := a.GetChannel(post.ChannelId)
+	if err != nil {
+		return 0, err
+	}
+
+	keywords := addMentionKeywordsForUser(
+		map[string][]string{},
+		user,
+		map[string]string{},
+		&model.Status{Status: model.STATUS_ONLINE}, // Assume the user is online since they would've triggered this
+		true, // Assume channel mentions are always allowed for simplicity
+	)
+
+	posts, nErr := a.Srv().Store.Thread().GetPosts(post.Id, timestamp)
```

Comment:
This begs the question: do we need to retrieve the whole posts just to get a count? Could the cases below be handled in pure SQL?

Reply:
@streamer45 - we need the post contents to extract the mentions. can't do it with pure sql

Reply:
I see. Got the wrong impression from the direct channel case which doesn't make use of messages.
Should mentions live in their own table then? Never mind, I'll stop here :grin: